### PR TITLE
fix(server): remove regexp check from method factory

### PR DESCRIPTION
**Pull request by hekike**
Closes

https://github.com/restify/node-restify/issues/1644

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -1484,8 +1486,24 @@ function errEvtNameFromError(err) {
...
+     * @example
+     * server.get('/example/near/:lat-:lng/radius/:r', (req, res, next) => {});
+     * server.get({ path: '/foo' }, (req, res, next) => {});
+     * server.get({ url: '/foo' }, (req, res, next) => {});
+     */
     return function serverMethod(opts) {
-        if (opts instanceof RegExp || typeof opts === 'string') {
+        // RegExp deprecation message
+        if (opts instanceof RegExp) {
+            new TypeError(REGEXP_PATH_DEPRECATED_TEXT);
```
Comment (DonutEspresso) [inline lib/server.js]: Is it deprecated or completely broken? Did you mean to throw this error?

Comment (hekike) [reply lib/server.js]: It's completely removed in 7x.

Comment (DonutEspresso) [reply lib/server.js]: Cool, then shouldn't we throw here then?

Comment (hekike) [reply lib/server.js]: oops, that was the plan :D

**Comment by hekike** [comment]
@DonutEspresso I addressed your comments.

**Pull request closed without merging**